docs/openhabian-amanda.md

```diff
@@ -564,13 +567,90 @@ You could also move that temporary recovered image file to your Windows PC that
+drwxrwxr-x openhabian/openhabian 16 2020-02-09 21:16 ./.vscode/
+drwxrwxr-x openhab/openhab       25 2020-01-27 21:53 ./html/
+drwxrwxr-x openhab/openhab       10 2020-01-27 21:52 ./icons/
+drwxrwxr-x openhab/openhab      884 2020-04-13 20:30 ./icons/classic/
+drwxrwxr-x openhab/openhab       25 2020-01-28 18:03 ./items/
+drwxrwxr-x openhab/openhab       13 2020-01-27 21:53 ./persistence/
+drwxrwxr-x openhab/openhab       25 2020-01-28 18:04 ./rules/
+drwxrwxr-x openhab/openhab       26 2020-01-28 18:04 ./scripts/
+drwxrwxr-x openhab/openhab       53 2020-01-27 21:53 ./services/
+drwxrwxr-x openhab/openhab       27 2020-01-28 18:05 ./sitemaps/
+drwxrwxr-x openhab/openhab       28 2020-01-27 21:52 ./sounds/
+drwxrwxr-x openhab/openhab       26 2020-01-28 18:05 ./things/
+drwxrwxr-x openhab/openhab       40 2020-01-28 18:05 ./transform/
+...
+-rwxrwxrwx openhabian/openhabian 13063 2020-04-08 12:05 ./sitemaps/demo.sitemap
+-rw-rw-r-- openhab/openhab         241 2020-01-13 08:53 ./sitemaps/readme.txt
+-rw-rw-r-- openhab/openhab        8776 2020-01-13 08:53 ./sounds/barking.mp3
+-rw-rw-r-- openhab/openhab      102399 2020-01-13 08:53 ./sounds/doorbell.mp3
+```
+
+```
+[14:11:52] root@openhabianpi:/volatile/backup/slots/slot14# dd if=00002.openhab._etc_openhab2.0 bs=32k skip=1 | tar xvf -
+./
+./.vscode/
+./html/
+./icons/
+...
+./transform/de.map
+./transform/en.map
+./transform/readme.txt
+```
+
+You can then put the extracted files back in the correct location on your openHABian box. You do not even have to put back the amanda_data files since Amanda will just build new indexes based on the new daily backups, just like it did the first time Amanda was installed. It is highly recommended that you make a backup copy of the openHABian box SD card as soon as it is fully back in operation. You may have had to reinstall SNAPSHOT.jar binding files, reconfigure exim4 to receive the Amanda backup emails and other settings or files not backed up by default by Amanda. To do so, you can include `/dev/mmcblk0` in the disklist on your next backup run and restore that to a new SD card (see restore section). If your SD card is large then daily backups of the SD card might be a problem. In addition, you may prefer to be able to just put a replacement SD card in your openHABian box almost immediatley should you have a failed SD card crash. There are several options:
+1. Shutdown the working openHABian box, take out the SD card and make a backup copy on another computer using, for example
+    `andrew@Meerkat:~/rp3_backup$ sudo dd if=/dev/sdc of=raspbian_backup19April2020.img`
+2. Make a copy of the openHABian SD card on your storage device using
+    `[13:44:51] openhabian@openhab:~$ sudo dd if=/dev/mmcblk0 of=/storage/server/sdcard-backups/sd_card_$(date +\%Y\%m\%d\%H\%M\%S).img`
+3. Automate this process, where SD card backups are made to your storage device on the first day of every month, by adding the following to the `/etc/cron.d/amanda` file
```

Comment:
I don't support these recommendations #1 and #3 
#2 can do what #1 does without downtime. You could rephrase to only use #1 if OH is already down and #2 when it's running, but since we only want to copy a _working_ OH, we shouldn't suggest that and only propose #2 

On recommendation #3 there's a lot of things that can go wrong.
Better to add a second Amanda config with disklist to only contain `/etc/mmcblk0`. You said you already did or do.
So could you describe the steps you did to add that (starting from a copy of the installed config) ?
Then put that paragraph in a suitable location somewhere up in the Installation section, and just reference it here.

Reply:
I do not know your openhabian installation. My perspective, and that of others on the forums, is that when the SD card fails (and they do all too often) there is no openhab, or openhabian, and no Amanda. A backup SD card is required first. I looked into disklist and a second config. The 32 GB SD card size swamps everything else and I came to the conclusion that a dd to storage (such as my NAS) was better and Amanda would be doing basically the same thing backup wise with `/dev/mmcblk0` in the disklist, but making access to the SD card backup much harder. What is your concern re #3? A very similar cron entry is used to make the amanda_data backups. I will look into adding some of this info earlier on in the doc.

Reply:
If you read the docs from the beginning you'll notice a recommendation _to use Amanda_ to create an SD image right after openhabian initial installation.
So you will have a working openHABian+Amanda at hand.
If you didn't follow the advice, you can also flash openHAbian image and install Amanda from there, then copy back your index and backup files.
This is what you propose to do again here, but using fractional scripts via cron. That does not make sense. No it's neither better nor easier to use dd or whatever. I fully disagree with that statement of yours.
Feel free to handle it like that for yourself but I won't put that in the docs, effectively recommending this to everyone. For sure I don't want Amanda docs to contain recommendations how to perform the same functions without Amanda.

Re: #3 you recommend to manually implement what Amanda can do for you but it can do better as it automates that process and it is way more reliable in doing so than any shell script.
So let Amanda handle this.

Reply:
I understand that you want people to use Amanda. Your solution for a dead SD card crash is for the user to use an "old" SD card backup to then use Amanda to restore a more recent SD card backup (which is unlikely to exist if users have followed the guide and not backed up a SD card greater than 16 GB in size). They then flash a SD card with this more recent SD card backup and startup their openHABian box a second time. My proposal simply lets users know they can skip one SD card flash step and go straight to the most recent SD card backup. In both my and your scenarios the user still gets the latest amanda_data files back and then uses Amanda to get whatever else they need to restore their system. I have tried to improve the wording to explain this.

Reply:
> Your solution for a dead SD card crash is for the user to use an "old" SD card backup to then use Amanda to restore a more recent SD card backup

No. If the backup is not too old, it is the only one you need to restore to SD. Then copy back the Amanda database and dist-upgrade to turn it into an up-to-date image, finally restore openHAB config.
This is not two restores but one, it is less work, more fool-proof and more reliable than your proposal.
IMO there's 2 type of users: most are Windows people or Linux novices who don't know to handle dd and who do not dare to touch anything Linux that they do not understand, even more so when they're under stress because their smart home server is failing. These you must present with an easy, fool-proof method to restore the SD image. That's what Amanda is. If they do not understand the UNIX command that are already written in this section then this is GOOD as it'll make them refrain from trying without possessing the knowledge that are required to apply these commands. 
The remaining people are Linux experts that already know what to do by looking at the existing docs. No additional need to teach those.
Who you describe another solution for is kind of a third group with mediocre knowledge inbetween those two I named ... but that group is of marginal size, I would even argue it does not exists and my two groups already cover all users.


>(which is unlikely to exist if users have followed the guide and not backed up a SD card greater than 16 GB in size).

You have a point here. But if people dont have a backup SD image or a too large card then there is a need to take care of that at the root of the problem.
It actually isn't a problem to restore 32 or 64 GB cards, it's just a waste of ressources and takes much more time than it takes for a 16GB card.
If you carefully read the recommendation in the existing docs to create a backup SD card right after openHAB was successfully installed for the first time, there is NO space limitation to apply to that recommendation.
That would possibly need to be reworded (a good job for a native speaker like you) and presented in a more prominent manner and explicitly pointed out that it should be done at least once for large cards, too.

As said the easiest and most reliable way to create a SD backup card (or to do it at regular intervals) is to setup another Amanda config to backup /dev/mmcblk0 maybe once every one or two months,and best is to automate that. NOTE: the existing text on backup assumes DAILY runs. 32/64GB are too much to backup that _every day_, yes. But backing it up once a month OTOH is fine.
So please describe how to setup another Amanda config (starting from a copy of the existing one) and describe the steps to do so. Or create the code to do it.
Finally just refer to that procedure in the "things gone wrong" section.

Reply:
I had a quick look at backup.bash and it looks like it would be straightforward to add some lines to the cronjob section to add the monthly SD card backup that I suggested. Users would only need to know how to write a new SD card from their backup device and they would automatically have an SD card backup that would be a maximum of one month old. I could implement this and modify and review the wording of the documentation to cover this and general English clarity. I think this is a better solution than adding another Amanda config. What do you think?

Reply:
Sorry but no way.
There's a fixed, important distriction between openHABian to take care of system backup and openhab-cli/backup.bash which is part of the openhab-linuxpkg repo and only meant to backup openHAB config.
Another Amanda config is also way simpler and more reliable.

Reply:
Sorry if that now sounds harsh but stop bringing up that same solution of yours again and again. Another Amanda config is simpler and way more reliable so I won't accept code to circumvent Amanda. My rationale is given above, agree with that or not.
You can produce a 2nd Amanda config along the same lines in backup.bash.

Reply:
Quite a long time ago, before installing Amanda, I did try log2ram, with zero success. ZRAM has been on my list of action items. At the moment I have been buried in other activities and hope to get back to openHABian soon...